hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/ReadableAccountStore.java

```diff
@@ -133,17 +185,70 @@ private Long getAccountNum(final AccountID id) {
             final var ret = aliases.get(alias.toStringUtf8());
             return ret == null ? MISSING_NUM : ret;
         }
-        return id.getAccountNum();
+        return idOrAlias.getAccountNum();
     }
 
-    private KeyOrLookupFailureReason validateKey(final JKey key) {
-        if (key == null) {
-            throw new IllegalArgumentException("Provided Key is null");
+    /**
+     * Returns the contract leaf for the given contract id. If the contract doesn't exist returns
+     * {@code Optional.empty()}
+     *
+     * @param id given contract number
+     * @return merkle leaf for the given contract number
+     */
+    private Optional<HederaAccount> getContractLeaf(@NonNull final ContractID id) {
+        final var contractNum = getContractNum(id);
+        if (contractNum.equals(MISSING_NUM)) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(accountState.get(contractNum));
+    }
+
+    /**
+     * Gets contract's number if the provided contract id is an evm address. If not, returns the
+     * contract's number
+     *
+     * @param idOrAlias provided account id
+     * @return account number
+     */
+    private Long getContractNum(final ContractID idOrAlias) {
+        if (isAlias(idOrAlias)) {
+            final var alias = idOrAlias.getEvmAddress();
+            final var evmAddress = alias.toByteArray();
+            if (isMirror(evmAddress)) {
+                return numOfMirror(evmAddress);
+            }
+            var entityNum = aliases.get(alias.toStringUtf8());
```

Comment:
What's the motivation for `toStringUtf8()` here and L#228?

Reply:
That was added in `StatesAPI` PR. I think, because key in `ReadableStates` or `WritableStates`should be comparable. But `ByteString` is not. So  using `String` for now.